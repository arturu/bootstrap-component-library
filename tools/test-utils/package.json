{
  "name": "@openeuropa/bcl-test-utils",
  "version": "0.1.3",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
