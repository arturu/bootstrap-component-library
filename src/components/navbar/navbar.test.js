import { merge, renderTwigFileAsNode } from "@oe/test-utils";

const template = "@oe/navbar/navbar.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - navbar", () => {
  ["light", "dark"].forEach((color_set) => {
    test(`${color_set} renders correctly`, () => {
      expect.assertions(1);

      return expect(render({ color_set })).resolves.toMatchSnapshot();
    });
  });
});
