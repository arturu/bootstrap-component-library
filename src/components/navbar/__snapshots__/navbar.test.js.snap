// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - navbar dark renders correctly 1`] = `
<test>
  <nav
    class="oe-navbar navbar navbar-dark navbar-expand-lg"
  >
    <div
      class="container"
    >
      <button
        aria-controls="navbarNavDropdown"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-bs-target="#navbarNavDropdown"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarNavDropdown"
      >
        <ul
          class="navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="nav-link active"
              href="#"
            >
              Home
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#"
            >
              Item 01
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#"
            >
              Item 02
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              aria-expanded="false"
              class="nav-link dropdown-toggle"
              data-bs-toggle="dropdown"
              href="#"
              id="navbarDropdownMenuLink"
              role="button"
            >
              
            Dropdown link
          
            </a>
            <ul
              aria-labelledby="navbarDropdownMenuLink"
              class="dropdown-menu"
            >
              <li>
                <a
                  class="dropdown-item"
                  href="#"
                >
                  Action
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="#"
                >
                  Another action
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="#"
                >
                  Something else here
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</test>
`;

exports[`OE - navbar light renders correctly 1`] = `
<test>
  <nav
    class="oe-navbar navbar navbar-light navbar-expand-lg"
  >
    <div
      class="container"
    >
      <button
        aria-controls="navbarNavDropdown"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-bs-target="#navbarNavDropdown"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarNavDropdown"
      >
        <ul
          class="navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="nav-link active"
              href="#"
            >
              Home
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#"
            >
              Item 01
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#"
            >
              Item 02
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              aria-expanded="false"
              class="nav-link dropdown-toggle"
              data-bs-toggle="dropdown"
              href="#"
              id="navbarDropdownMenuLink"
              role="button"
            >
              
            Dropdown link
          
            </a>
            <ul
              aria-labelledby="navbarDropdownMenuLink"
              class="dropdown-menu"
            >
              <li>
                <a
                  class="dropdown-item"
                  href="#"
                >
                  Action
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="#"
                >
                  Another action
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="#"
                >
                  Something else here
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</test>
`;
